sky/utils/cli_utils/status_utils.py

```diff
@@ -316,3 +301,143 @@ def _get_estimated_cost_for_cost_report(
         return '-'
 
     return f'$ {cost:.2f}'
+
+
+def show_kubernetes_cluster_status_table(clusters: List[Any],
+                                         show_all: bool) -> None:
+    """Compute cluster table values and display for Kubernetes clusters."""
+    status_columns = [
+        StatusColumn('USER', lambda c: c['user']),
+        StatusColumn('NAME', lambda c: c['cluster_name']),
+        StatusColumn(
+            'LAUNCHED',
+            lambda c: log_utils.readable_time_duration(c['launched_at'])),
+        StatusColumn('RESOURCES',
+                     lambda c: c['resources_str'],
+                     trunc_length=70 if not show_all else 0),
+        StatusColumn('STATUS', lambda c: c['status'].colored_str()),
+        # TODO(romilb): We should consider adding POD_NAME field here when --all
+        #  is passed to help users fetch pod name programmatically.
+    ]
+
+    columns = [
+        col.name for col in status_columns if col.show_by_default or show_all
+    ]
+    cluster_table = log_utils.create_table(columns)
+
+    # Sort table by user, then by cluster name
+    sorted_clusters = sorted(clusters,
+                             key=lambda c: (c['user'], c['cluster_name']))
+
+    for cluster in sorted_clusters:
+        row = []
+        for status_column in status_columns:
+            if status_column.show_by_default or show_all:
+                row.append(status_column.calc(cluster))
+        cluster_table.add_row(row)
+
+    if clusters:
+        click.echo(f'{colorama.Fore.CYAN}{colorama.Style.BRIGHT}'
+                   f'SkyPilot clusters'
+                   f'{colorama.Style.RESET_ALL}')
+        click.echo(cluster_table)
+    else:
+        click.echo('No existing SkyPilot clusters on Kubernetes.')
```

Comment:
"No SkyPilot resources found in the active Kubernetes context."

(not just clusters)